test/cases/orchestration.py

```diff
@@ -506,37 +507,45 @@ def down(self):
 
     @staticmethod
     def has_coverage_data():
-        command = docker_compose_command(
-            "exec",
-            "-T",
-            "--",
-            "nginx",
-            "bash",
-            "-c",
-            "find /tmp -name '*.profraw' -print -quit | grep -q .",
-        )
-        result = subprocess.run(command)
-        return result.returncode == 0
+        command = docker_compose_command("exec", "-T", "--", "nginx", "find",
+                                         "/tmp", "-name", "*.profraw")
+        result = subprocess.run(command, capture_output=True)
+        if result.returncode != 0:
+            return False
+
+        return len(result.stdout)
 
     @staticmethod
     def create_coverage_tarball():
-        tar_command = docker_compose_command(
+        cmd = docker_compose_command(
             "exec",
             "-T",
             "--",
             "nginx",
-            "bash",
-            "-c",
-            "tar --transform='s@tmp/@@' -czf - -T <(find /tmp -maxdepth 1 -name '*.profraw')",
+            "find",
+            "/tmp",
+            "-name",
+            "*.profraw",
```

Comment:
I think it would be simpler to just avoid process substitution and pass the files to tar on the command line:

```python
tar_command = docker_compose_command(
    "exec",
    "-T",
    "--",
    "nginx",
    "sh",
    "-c",
    "find /tmp -maxdepth 1 -name '*.profraw' | xargs tar --transform='s@tmp/@@' -czf -",
)
```

Reply:
That's a possibility. It's clever, however, I am afraid not everyone can understand this command line, whereas python code is much more expressive.